.github/workflows/release.yml

```diff
@@ -0,0 +1,89 @@
+name: Build releases
+
+on:
+  workflow_dispatch:
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        include:
+          - os: macos-latest-large
+            target: aarch64-apple-darwin
+          - os: macos-latest-large
+            target: aarch64-apple-ios
+          - os: macos-latest-large
+            # cross-compile for Linux ARM using Apple Silicon until we have ARM runners
+            target: aarch64-unknown-linux-musl
+          - os: macos-latest
+            target: x86_64-apple-darwin
+          - os: ubuntu-latest
+            target: x86_64-unknown-linux-musl
+          # - os: windows-latest
+          #   target: x86_64-pc-windows-msvc
+    name: Build ${{matrix.target}}
+    runs-on: ${{matrix.os}}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Get short SHA
+        id: sha
+        run: echo "sha_short=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+
+      - name: Install build tools (Linux)
+        if: ${{matrix.os == 'ubuntu-latest'}}
+        run: |
+          sudo apt-get update
+          sudo apt-get install musl-tools
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@1.70.0
```

Comment:
Why fix on a specific Rust version? IMHO should matrix the MSRV version and stable.